--- working.SDK.UBNT.v5.5/openwrt/package/quagga-unstable/Makefile.orig	2012-08-13 22:00:07.000000000 +0200
+++ working.SDK.UBNT.v5.5/openwrt/package/quagga-unstable/Makefile	2012-08-13 22:36:22.387551771 +0200
@@ -177,11 +177,11 @@
 
 CONFIGURE_ARGS+= \
 	--localstatedir=/var/run/quagga \
-	--sysconfdir=/etc/quagga/ \
+	--sysconfdir=/etc/persistent/ \
 	--enable-shared \
 	--disable-static \
-	--enable-user=network \
-	--enable-group=network \
+        --enable-user=ubnt \
+        --enable-group=admin \
 	--enable-pie=no \
 	--enable-multipath=8 \
 	--disable-ospfclient \
@@ -204,7 +204,6 @@
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) ./files/quagga $(1)/usr/sbin/quagga.init
 	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_BIN) ./files/quagga.init $(1)/etc/init.d/quagga
 endef
 
 define Package/quagga-watchquagga/install
@@ -215,65 +214,41 @@
 define Package/quagga-zebra/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/zebra $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/zebra.conf
 endef
 
 define Package/quagga-bgpd/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/bgpd $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/bgpd.conf
 endef
 
 define Package/quagga-isisd/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/isisd $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/isisd.conf
 endef
 
 define Package/quagga-ospfd/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/ospfd $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/ospfd.conf
 endef
 
 define Package/quagga-ospf6d/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/ospf6d $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/ospf6d.conf
 endef
 
 define Package/quagga-ripd/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/ripd $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/ripd.conf
 endef
 
 define Package/quagga-ripngd/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/ripngd $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/ripngd.conf
 endef
 
 define Package/quagga-babeld/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/babeld $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc/quagga
-	chmod 0750 $(1)/etc/quagga
-	$(INSTALL_CONF) ./files/quagga.conf $(1)/etc/quagga/babeld.conf
 endef
 
 define Package/quagga-vtysh/install
